### PR TITLE
add TableMaster table recognition model

### PR DESCRIPTION
**Pull request by tonytonglt**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/data/transforms/table_transform.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+        mean = mean if mean is not None else [0.485, 0.456, 0.406]
+        std = std if std is not None else [0.229, 0.224, 0.225]
+
+        shape = (3, 1, 1) if order == "chw" else (1, 1, 3)
+        self.mean = np.array(mean).reshape(shape).astype("float32")
+        self.std = np.array(std).reshape(shape).astype("float32")
+
+    def __call__(self, data):
+        img = data["image"]
+        from PIL import Image
```
Comment (chengtianwu) [inline mindocr/data/transforms/table_transform.py:386]: 放到最上面

File: configs/table/README_CN.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+python tools/train.py --config configs/table/table_master.yml
+```
+
+* 分布式训练
+
+请确保yaml文件中的`distribute`参数为True。
+
+```shell
+# n is the number of GPUs/NPUs
+mpirun --allow-run-as-root -n 8 python tools/train.py --config configs/table/table_master.yml
```
Comment (chengtianwu) [inline configs/table/README_CN.md:145]: table_master.yaml

File: mindocr/models/heads/table_master_head.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        self.linears = clones(nn.Dense(d_model, d_model), 4)
+        self.dropout = nn.Dropout(p=dropout)
+        self.relu = nn.ReLU()
+        self.batch_matmul = ops.BatchMatMul()
+
+    def construct(self, query, key, value, mask=None):
+        B = query.shape[0]
+
+        # 1) Do all the linear projections in batch from d_model => h x d_k
+        # print(self.linears)
```
Comment (panshaowu) [inline mindocr/models/heads/table_master_head.py:228]: 无用代码删除

File: mindocr/models/heads/table_master_head.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+        normed_x = self.norms[0](x)
+        x = x + self.dropouts[0](self.self_attn(normed_x, normed_x, normed_x, tgt_mask))
+        normed_x = self.norms[1](x)
+        x = x + self.dropouts[1](self.src_attn(normed_x, feature, feature, src_mask))
+        normed_x = self.norms[2](x)
+        x = x + self.dropouts[2](self.feed_forward(normed_x))
+        return x
+
+
+class MultiHeadAttention(nn.Cell):
```
Comment (panshaowu) [inline mindocr/models/heads/table_master_head.py:212]: 类似的Transformer代码最好复用已有的实现，或者抽取为公共代码。

**Review by panshaowu — APPROVED** [review]

**Review by chengtianwu — APPROVED** [review]

**Pull request merged**